### PR TITLE
Add remaining KComponent interfaces and implementations.

### PR DESCRIPTION
**Pull request by markusthoemmes**
## Proposed Changes

As per title, this concludes the interface implementations and makes both types conform to the entire `KComponent` interface.

**Release Note**

```release-note
NONE
```

/assign @aliok @houshengbo @jcrossley3

### CODE REVIEW

**Comment by markusthoemmes** [comment]
/retest

I'd argue the coverage warning is benign.

File: pkg/apis/operator/v1alpha1/common.go [hunk 1]
```diff
@@ -33,6 +35,27 @@ const (
...
+	// GetSpec returns the common spec for all known types.
+	GetSpec() KComponentSpec
+	// GetStatus returns the common status of all known types.
+	GetStatus() KComponentStatus
+}
+
+// KComponentSpec is a common interface for accessing the common spec of all known types.
+type KComponentSpec interface {
+	// GetConfig returns means to override entries in upstream configmaps.
+	GetConfig() map[string]map[string]string
```
Comment (jcrossley3) [inline pkg/apis/operator/v1alpha1/common.go:52]: nit: i think i'd like to typedef this to `ConfigMapData` just to clarify its purpose and reduce the noise.

File: pkg/apis/operator/v1alpha1/common.go [hunk 2]
```diff
@@ -78,6 +101,21 @@ type CommonSpec struct {
 	Resources []ResourceRequirementsOverride `json:"resources,omitempty"`
 }
 
+// GetConfig implements KComponentSpec.
+func (c *CommonSpec) GetConfig() map[string]map[string]string {
```
Comment (jcrossley3) [inline pkg/apis/operator/v1alpha1/common.go:105]: ibid

**Comment by markusthoemmes** [comment]
@jcrossley3 implemented your suggestion

**Pull request merged**